File: cpp/2020/sept/sum-root-to-leaf.cpp
int sumRootToLeaf(TreeNode *r, int val = 0) {
    if (r == nullptr) return 0;
    val = ((val << 1) + r->val) % 1000000007;
    return (r->left == r->right ? val : sumRootToLeaf(r->left, val) + sumRootToLeaf(r->right, val)) % 1000000007;
}

File: archive/go/2020/sept/sum-root-to-leaf.go
package sept

import "leetcode/common"

func sumRootToLeaf(root *common.TreeNode) int {
	return dfs(root, 0)
}

func dfs(root *common.TreeNode, val int) int {
	if root == nil {
		return 0
	}
	val = val*2 + root.Val
	if root.Left == root.Right {
		return val
	}
	return dfs(root.Left, val) + dfs(root.Right, val)
}
